test/androidTest/java/org/thoughtcrime/securesms/EspressoHelper.java

```diff
@@ -0,0 +1,250 @@
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.Espresso.pressBack;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.longClick;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition;
+import static android.support.test.espresso.matcher.ViewMatchers.isAssignableFrom;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+enum Location {
+    HOME,
+    CONVERSATION,
+    MESSAGE_SELECTED,
+    PINNED,
+}
+
+public class EspressoHelper {
+    private String phoneNumber;
+    private Location location;
+
+    public EspressoHelper(ActivityTestRule<ConversationListActivity> activityRule) {
+        activityRule.launchActivity(new Intent());
+
+        openActionBarOverflowOrOptionsMenu(InstrumentationRegistry.getTargetContext());
+
+        onView(withText(R.string.text_secure_normal__menu_settings))
+                .perform(click());
+
+        final String[] stringHolder = { null };
+        onView(withId(R.id.number)).perform(new ViewAction() {
+            @Override
+            public Matcher<View> getConstraints() {
+                return isAssignableFrom(TextView.class);
+            }
+
+            @Override
+            public String getDescription() {
+                return "getting text from a TextView";
+            }
+
+            @Override
+            public void perform(UiController uiController, View view) {
+                TextView tv = (TextView)view; //Save, because of check in getConstraints()
+                stringHolder[0] = tv.getText().toString();
+            }
+        });
+
+        this.phoneNumber = stringHolder[0];
+
+        pressBack();
+
+        this.location = Location.HOME;
+    }
+
+    public void done() {
+        if (this.location != Location.HOME) {
+            this.goHome();
+        }
+    }
+
+    /** ASSERTIONS */
+
+    public EspressoHelper findId (int id) {
+        onView(withId(id))
+                .check(matches(isDisplayed()));
+
+        return this;
+    }
+
+    public EspressoHelper findText (String text) {
+        onView(withText(text))
+                .check(matches(isDisplayed()));
+
+        return this;
+    }
+
+    public EspressoHelper not_findText (String text) {
+        try {
+            onView(withText(text))
+                    .check(matches(isDisplayed()));
+
+            throw new Error("EspressoHelper.not_findText: View found matching \"" + text + "\"");
+        } catch (NoMatchingViewException e) {}
```

Comment:
Agreed ^

Reply:
The whole point is that the error SHOULD be thrown because it is a negative check. So I am only catching that exact error, and anything else will fail the test